### ISSUE TITLE
Snapshot generation fails for Task resources

### ISSUE THREAD

**Issue by wetret** [highmed/highmed-dsf#43]
Snapshot generation fails if a task profile has more defined input slices than the highmed-task-base profile and throws the following exception:

```
org.hl7.fhir.exceptions.DefinitionException: Differential does not have a slice: 
Task.input/ (b:44 of 55 / 1/ 11) in profile 
```

**Comment by hhund** [comment]
Here's a patch that should fix issues with highmed-task-base-0.1.0.xml and highmed-task-request-simple-feasibility-0.1.0.xml I'm not sure tho if all my min max additions are necessary. The important bits are the <slicing /> tags for task.input and task.output

```
diff --git a/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-base-0.1.0.xml b/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-base-0.1.0.xml
index 75c3164..20286d0 100644
--- a/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-base-0.1.0.xml
+++ b/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-base-0.1.0.xml
@@ -52,7 +52,7 @@
           <type value="value" />
           <path value="type.coding.code" />
         </discriminator>
-        <rules value="openAtEnd" />
+        <rules value="open" />
       </slicing>
       <min value="3" />
     </element>
@@ -145,6 +145,11 @@
       <path value="Task.output" />
       <sliceName value="error" />
     </element>
+    <element id="Task.output:error.type.coding">
+      <path value="Task.output.type.coding" />
+      <min value="1" />
+      <max value="1" />
+    </element>
     <element id="Task.output:error.type.coding.system">
       <path value="Task.output.type.coding.system" />
       <fixedUri value="http://highmed.org/fhir/CodeSystem/bpmn-message" />
diff --git a/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-request-simple-feasibility-0.1.0.xml b/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-request-simple-feasibility-0.1.0.xml
index acd02ba..6007b4d 100644
--- a/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-request-simple-feasibility-0.1.0.xml
+++ b/dsf-fhir/dsf-fhir-server/src/main/resources/fhir/StructureDefinition/highmed-task-request-simple-feasibility-0.1.0.xml
@@ -18,7 +18,15 @@
     </element>
     <element id="Task.input">
       <path value="Task.input" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="type.coding.code" />
+        </discriminator>
+        <rules value="closed" />
+      </slicing>
       <min value="4" />
+      <max value="4" />
     </element>
     <element id="Task.input:message-name">
       <path value="Task.input" />
@@ -34,6 +42,11 @@
       <min value="1" />
       <max value="1" />
     </element>
+    <element id="Task.input:research-study.type.coding">
+      <path value="Task.input.type.coding" />
+      <min value="1" />
+      <max value="1" />
+    </element>
     <element id="Task.input:research-study.type.coding.system">
       <path value="Task.input.type.coding.system" />
       <fixedUri value="http://highmed.org/fhir/CodeSystem/task-input" />
@@ -49,9 +62,27 @@
         <targetProfile value="http://highmed.org/fhir/StructureDefinition/highmed-research-study" />
       </type>
     </element>
+    <element id="Task.output">
+      <path value="Task.output" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="type.coding.code" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <min value="2" />
+    </element>
     <element id="Task.output:participatingMedics">
       <path value="Task.output" />
       <sliceName value="participatingMedics" />
+      <min value="1" />
+      <max value="1" />
+    </element>
+    <element id="Task.output:participatingMedics.type.coding">
+      <path value="Task.output.type.coding" />
+      <min value="1" />
+      <max value="1" />
     </element>
     <element id="Task.output:participatingMedics.type.coding.system">
       <path value="Task.output.type.coding.system" />
@@ -70,6 +101,13 @@
     <element id="Task.output:cohortSize">
       <path value="Task.output" />
       <sliceName value="cohortSize" />
+      <min value="1" />
+      <max value="1" />
+    </element>
+    <element id="Task.output:cohortSize.type.coding">
+      <path value="Task.output.type.coding" />
+      <min value="1" />
+      <max value="1" />
     </element>
     <element id="Task.output:cohortSize.type.coding.system">
       <path value="Task.output.type.coding.system" />
```

**Comment by hhund** [comment]
I not sure yet but the proposed fix e9e0fff for #43 might be a workaround for #45

**Comment by wetret** [comment]
According to the FHIR specification a re-slice of an already sliced element in the parent profile needs to define the <slicing> tag again in the element to be sliced. See https://www.hl7.org/fhir/profiling-examples.html, section 5.1.1.7. This means that Forge probably has an error in creating slices in child profiles since the <slicing> information is missing.
